### DISCUSSION TITLE
Custom feature display, layout, interfaces how to?

### DISCUSSION THREAD

**Discussion by markgiul** [directus/directus#4878] [Archive]
Hello, I need to create a custom layout I followed the guide [https://docs.directus.io/guides/layouts/](https://docs.directus. io/guides/layouts/), ok for the build part, but what I can't understand is how to physically develop the component, which in the case of the guide is very basic (and by the way I can't get it loaded in the directus instance), but if I want to interact for example in the case of the layouts (which should allow me to create a custom view for the edit part of a collection) with all the interfaces and data that will pass in that form? Do I have to create a special project with all the dependencies, or is there another way that escapes me at the moment?
Thank you very much for any suggestions. 
P.S sorry for my bad English

**Comment by philleepflorence** [comment]
What steps of the Guide didn't work as expected?
See if the Props listed in the [Guide](https://docs.directus.io/guides/layouts/#src-layout-vue) have the data you need within the layout.
For dependencies, are you referring to the package.json dependencies?

**In the meantime, the following may help with building a custom layout extension.**

> Tip
> A basic understanding of [Vue](https://v3.vuejs.org/guide/introduction.html) and Vue Components is necessary for extending Directus Vue Components; Introduction to [ES Modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) and [Rollup](https://rollupjs.org/guide/en/#overview) is helpful!  

- Have a local instance of Directus running for development, see the [Running Locally Documentation](https://docs.directus.io/contributing/running-locally/); _you can watch for changes and build as you develop_.
- Get familiar with [Directus Vue Components](https://docs.directus.io/reference/app/components/).
- Identify the core layout that is closest to your concept in the [Directus Documentation Concept - Core Layouts](https://docs.directus.io/concepts/layouts/) and locate the Vue Component Directory and Files in the [Directus GitHub Repository](https://github.com/directus/directus/tree/main/app/src/layouts) to use as a template guide and reference.
- Follow the Guide to Creating Custom Layouts and then copy the compiled layout to the local Directus instance.

Additional confirmation on the steps may be needed based on the current D9 release.

**Reply by markgiul** [reply]
thanks, for this information, what i was mainly interested in is to understand how to develop, for example, a custom module by accessing the internal components (api, collection configure and etc..) of directus from a version other than the development version, but apparently you have to work on a local directus installation running in development.

**Reply by alextremblay** [reply]
According to the documentation for custom layouts, there is a prop that gets passed in to a custom layout called collection, but it's only the name of the collection, not the actual collection (array of items). Can the docs be expanded to show how to access the data that a layout should display?

Also, there are docs on how to use the directus SDK outside of the directus app to authenticate to the API and fetch data, but i don't see anything in the App extension documentation for Modules, Layouts, Interfaces, or Displays detailing how to make API queries from inside of an extension within the app. Any extra info there would be greatly appreciated!

**Reply by rijkvanzanten** [reply]
@alextremblay You can use Vue's provide/inject to inject a field called `system` that contains a bunch of system internals you can use, including an authenticated Axios instance:

```js
export default {
  props: {
    collection: {
      type: String,
      required: true
    }
  },
  inject: ['system'],
  mounted() {
    this.system.api.get(`/items/${this.collection}`)
      .then((res) => console.log(res));
  }
}
```

(cc @nickrum )
